src/panels/lovelace/ha-panel-lovelace.ts

```diff
@@ -263,15 +264,22 @@ class LovelacePanel extends LitElement {
 
   private _checkLovelaceConfig(config: LovelaceConfig) {
     // Somehow there can be badges with value null, we remove those
-    config.views.forEach((view) => {
-      if (view.badges) {
-        view.badges = view.badges.filter(Boolean);
+    let checkedConfig;
+    config.views.forEach((view, index) => {
+      if (view.badges && !view.badges.every(Boolean)) {
+        checkedConfig = checkedConfig || {
+          ...config,
+          views: [...config.views],
+        };
+        checkedConfig.views[index] = { ...view };
+        checkedConfig.views[index].badges = view.badges.filter(Boolean);
       }
     });
+    return deepFreeze(checkedConfig || config);
```

Comment:
We shouldn't call `deepFreeze` on `config` again. That would have us traverse the whole tree for nothing.

Reply:
```suggestion
    return checkedConfig ? deepFreeze(checkedConfig) : config;
```